lib/graph3d/Graph3d.js

```diff
@@ -1680,6 +1646,42 @@ Graph3d.prototype._getColorsSize = function() {
 };
 
 
+/**
+ * Determine the color corresponding to a given value on the color scale.
+ *
+ * @param {number} [x] the data value to be mapped running from 0 to 1
+ * @param {number} [v] scale factor between 0 and 1 for the color brightness
+ * @returns {string}
+ * @private
+ */
+Graph3d.prototype._colormap = function(x, v=1) {
+
+  let r, g, b, a;
+  const colormap = this.colormap;
+  if (Array.isArray(colormap)) {
+    const maxIndex = colormap.length - 1;
+    const startIndex = Math.max(Math.floor(x * maxIndex), 0);
+    const endIndex = Math.min(startIndex + 1, maxIndex);
+    const innerRatio = x * maxIndex - startIndex;
+    const min = colormap[startIndex];
+    const max = colormap[endIndex];
+    r = min.r + innerRatio*(max.r - min.r);
+    g = min.g + innerRatio*(max.g - min.g);
+    b = min.b + innerRatio*(max.b - min.b);
+  } else if (typeof colormap === 'function') {
+    ({r, g, b, a} = colormap(x));
+  } else {
+    const hue = (1 - x) * 240;
+    ({r, g, b} = util.HSVToRGB(hue/360, 1, 1));
+  }
+  if (isNaN(a)) {
```

Comment:
Don't use isNaN unless you consider values such as `""` (yes, that's an empty string) to be valid numbers.

Reply:
What then? I don't think it matters here that much. The only cases that should occur here are `undefined`, `NaN`, or some number. The `isNaN` check was the easiest check I found to distungish these cases.

Reply:
```javascript
Number.isNaN(…)
```

Reply:
`Number.isNaN` is worse because it checks for identity with `NaN`, i.e. evaluates to `false` for `undefined`. Of course I could use something like `Number.isNaN(a + 0)` in which case the result should be the same as `isNaN`. To be honest, `isNaN` does pretty much exactly what we need here.

Reply:
Ok, if you want, we can go with `typeof a === 'number'` that should be fine as well.